src/components/ChangePassword/index.js

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { Text } from 'react-native';
+import { func } from 'prop-types';
+import { LOADING, SUCCESS } from '@rootstrap/redux-tools';
+
+import Modal from 'components/common/Modal';
+import Input from 'components/common/Input';
+import ErrorView from 'components/common/ErrorView';
+import Button from 'components/common/Button';
+import useChangePassword from 'hooks/useChangePassword';
+import useForm from 'hooks/useForm';
+import useTextInputProps from 'hooks/useTextInputProps';
+import strings from 'locale';
+import { BLACK } from 'constants/colors';
+import styles from './styles';
+
+const FIELDS = {
+  currentPassword: 'currentPassword',
+  password: 'password',
+  passwordConfirmation: 'passwordConfirmation',
+};
+
+const {
+  PASSWORD: { currentLabel, newLabel, confirmLabel, success },
+  COMMON: { done, cancel, ok },
+} = strings;
+
+const ChangePassword = ({ onHide }) => {
+  const { onSubmit, validator, error, status, reset } = useChangePassword();
+
+  const { values, errors, handleValueChange, handleSubmit, handleBlur } = useForm(
+    {
+      onSubmit,
+      validator,
+      validateOnBlur: true,
+    },
+    [onSubmit],
+  );
+
+  const inputProps = useTextInputProps(handleValueChange, handleBlur, values);
+
+  return (
+    <Modal>
+      {status !== SUCCESS ? (
+        <>
+          <Input
+            label={currentLabel}
+            error={errors[FIELDS.currentPassword]}
+            invalid={error || errors[FIELDS.currentPassword]}
+            {...inputProps(FIELDS.currentPassword)}
+            secureTextEntry
+            showErrorMessage
+          />
+          <Input
+            label={newLabel}
+            error={errors[FIELDS.password]}
+            invalid={error || errors[FIELDS.password]}
+            {...inputProps(FIELDS.password)}
+            secureTextEntry
+            showErrorMessage
+          />
+          <Input
+            label={confirmLabel}
+            error={errors[FIELDS.passwordConfirmation]}
+            invalid={error || errors[FIELDS.passwordConfirmation]}
+            {...inputProps(FIELDS.passwordConfirmation)}
+            secureTextEntry
+            showErrorMessage
+          />
+          <ErrorView errors={{ error }} />
+          <Button
+            handleOnPress={() => handleSubmit(values)}
+            additionalStyles={styles.done}
+            title={done}
+            isLoading={status === LOADING}
+          />
+          <Button handleOnPress={onHide} title={cancel} titleColor={BLACK} />
+        </>
+      ) : (
+        <>
+          <Text style={styles.success}>{success}</Text>
+          <Button
+            handleOnPress={() => {
+              reset();
+              onHide();
+            }}
+            additionalStyles={[styles.done, { width: 114 }]}
```

Comment:
You are setting `width: 157` on `styles.done` and then rewriting this with `width: 114`, I suppose you need only one of them, right?

Reply:
Yes. I just didn't want to add another style for the width, since the button uses the same properties as the done button

Reply:
I would create it anyways is better for mantainability. Tomorrow if we do a rebranding of the app we. just need to go to the styles file to fix things. This way inline styles might difficult a little bit our job. In and reduces the amount of styles being passeed by the react native bridge

Reply:
Ok, great! I'll do it. Thank you both!